micropy/project/modules/packages.py

```diff
@@ -119,92 +120,95 @@ def add_package(self, package, dev=False, **kwargs):
             dict: Dictionary of packages
 
         """
-        source = create_dependency_source(package)
+        self.log.debug(f"adding new dependency: {package}")
+        source = create_dependency_source(package, **kwargs)
         pkg = source.package
         self.log.info(f"Adding $[{pkg.name}] to requirements...")
         if self.packages.get(pkg.name, None):
             self.log.error(f"$[{pkg}] is already installed!")
             self.update()
             return None
-        self.packages[pkg.name] = pkg.pretty_specs
         try:
+            self.config.add(self.name + '/' + pkg.name, pkg.pretty_specs)
             self.load()
-        except ValueError:
+        except ValueError as e:
             self.log.error(f"Failed to find package $[{pkg.name}]!")
-            self.log.error("Is it available on PyPi?")
-            self.packages.pop(pkg.name)
-            self.parent.config.pop(f"{self.name}.{pkg}")
+            self.log.error("Is it available on PyPi?", exception=e)
+            self.config.pop(self.name + '/' + pkg.name)
         except Exception as e:
             self.log.error(
                 f"An error occured during the installation of $[{pkg.name}]!",
                 exception=e)
-            self.packages.pop(pkg.name)
-            self.parent.config.pop(f"{self.name}.{pkg}")
+            self.config.pop(self.name + '/' + pkg.name)
         else:
-            self.parent.config.set(f"{self.name}.{pkg}", pkg.pretty_specs)
             self.log.success("Package installed!")
         finally:
-            return self.packages
+            return self.config.raw()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [return statement in finally block may swallow exception](https://app.codacy.com/app/BradenM/micropy-cli/pullRequest?prid=4594550)